lib/internal/modules/esm/module_job.js

```diff
@@ -151,7 +167,33 @@ class ModuleJob {
     await this.instantiate();
     const timeout = -1;
     const breakOnSigint = false;
-    await this.module.evaluate(timeout, breakOnSigint);
+    try {
+      await this.module.evaluate(timeout, breakOnSigint);
+    } catch (e) {
+      if (e?.name === 'ReferenceError' &&
+          isCommonJSGlobalLikeNotDefinedError(e.message)) {
+        e.message += ' in ES module scope';
+
+        if (StringPrototypeStartsWith(e.message, 'require ')) {
+          e.message += ', you can use import instead';
+        }
+
+        const packageJSONUrl =
+          StringPrototypeStartsWith(this.module.url, 'file://') &&
+            RegExpPrototypeTest(/\.js(\?[^#]*)?(#.*)?$/, this.module.url) &&
+            require('internal/modules/esm/resolve')
+              .getCachedPackageScopeURL(this.module.url);
+        if (packageJSONUrl) {
+          e.message +=
+            `\n'${this.module.url}' cannot be parsed as an ES module, but ` +
```

Comment:
Perhaps let's leave out this URL as well?

Reply:
Agreed, the first sentence is not that useful anyway, let's drop it.